daceml/pytorch/module.py

```diff
@@ -0,0 +1,64 @@
+try:
+    import torch
+except ModuleNotFoundError:
+    raise ImportError("To use the torch frontend, please install torch")
+
+import os
+import tempfile
+from functools import wraps
+
+import torch.nn as nn
+import onnx
+
+from daceml.onnx import ONNXModel
+from daceml.onnx.shape_inference import infer_shapes
+
+
+class DACEModule(nn.Module):
+    def __init__(self, model, dummy_inputs=None):
+        super(DACEModule, self).__init__()
+
+        self.model = model
+        self.sdfg = None
+        if dummy_inputs is not None:
+            self.dace_model = self.initialize_sdfg(dummy_inputs)
+
+    def initialize_sdfg(self, dummy_inputs) -> ONNXModel:
+        export_fd, export_name = tempfile.mkstemp(suffix=".onnx")
+        infer_fd, infer_name = tempfile.mkstemp(suffix=".onnx")
```

Comment:
could use `with` statements instead of the `os.close`...

Reply:
I looked at this for a while and I'm not really sure how; `TemporaryFile` does have a context manager, but it returns a file object, which doesn't work with `onnx.load`.

Reply:
you can use the created object's `name` field to get the file path iirc